core/src/trezor/ui/text.py

```diff
@@ -29,35 +31,88 @@ def render_text(
     offset_x: int = TEXT_MARGIN_LEFT,
     offset_y: int = TEXT_HEADER_HEIGHT + TEXT_LINE_HEIGHT,
     offset_x_max: int = ui.WIDTH,
+    pre_linebreaked: bool = False,
 ) -> None:
+    if not pre_linebreaked:
```

Comment:
I don't like having both `pre_linebreaked` and `new_lines`. What is currently the difference? Could we maybe get rid of `new_lines` now?

Reply:
* both=`False`: `words` are treated as words, if a word fits on single line it is never broken, if it doesn't fit it's broken such that as many characters as possible stay on the first line. Spaces are treated as any other character when breaking a word containing them.
* `new_lines=True`: `words` are treated more as a lines, inserting `BR` after every one. Otherwise behaves as above. This is the most common use in our codebase.
* `pre_linebreaked=True`: `words` are expected to contain linebreaks at the right places and are not processed before rendering.

Thinking about this, the result will be the same often for the latter two options.

Then there's another behaviour we want but is not part of this PR - accept strings containing spaces and break on spaces wherever possible. Perhaps it would make sense to merge all these flags into something like `mode` which would take three different values?

Reply:
i'm still a little confused so I'm going to make a full table:

new_lines | pre_linebreaked | Behavior
--------- | --------------- | --------
False     | False           | Text is split at (a) explicit BRs, (b) end of screen
True      | False           | Text is split at (a) explicit BRs, (b) end of screen, (c) end of individual item. This is most common.
Any       | True            | Text is split at (a) explicit BRs. I.e, no magic happens if the text overflows.

Is this correct?

I believe that going forward, we will be able to completely get rid of the (c) variant, "end of individual item". But for this PR it stays.

I'm not sure, however, how valuable is keeping the (b) variant. If we removed that, then there is no difference in behavior between `pre_linebreaked=True` and `False.

Also implementation-wise it's just a matter of "reflowing" the `words` parameter via a separate call.

(BTW, I believe the reason `new_lines` is a parameter of `render_text()` is to save memory and/or processing time otherwise needed to insert the BR symbols to the list. But going forward we'll be splitting strings anyway so I don't think this is relevant, but let's keep that in mind)

Anyway.
I think `pre_linebreaked` should only be argument of `Text` class, not `render_text` call, and `Text` should be responsible for calling `break_lines`

also i'd call it something else, something along the lines of `auto_linebreaks=True` instead of `pre_linebreaked=False`

Reply:
Yeah, I'd say the table is correct. Agreed that it probably doesn't make sense to keep (b) long-term.

Regarding memory & time, this PR probably makes things worse but not sure how to get around that.

Will look into your suggestions, thanks!

Reply:
> Regarding memory & time, this PR probably makes things worse but not sure how to get around that.

Total allocations for device tests in `text.py` went from 180000 to 390000, that's pretty bad ...